test/docker_test.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```

Comment:
I assume the idea that this will not be run during Ctest (at end of cmake build), because it it probably a bit heavy for that?

Reply:
This is more of a validation test that "the docker image works", but IMHO it's pretty much unneeded.
If the application works and test passes, this will always work.

Reply:
Moreover, the `docker build` command already runs `cmake build` including the `system_test` during its build, so it's totally not intended to "run docker inside docker".